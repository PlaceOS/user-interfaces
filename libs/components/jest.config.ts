/* eslint-disable */
export default {
    displayName: 'components',
    preset: '../../jest.preset.js',
    setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
    globals: {},
    coverageDirectory: '../../coverage/libs/components',
    snapshotSerializers: [
        'jest-preset-angular/build/serializers/no-ng-attributes',
        'jest-preset-angular/build/serializers/ng-snapshot',
        'jest-preset-angular/build/serializers/html-comment',
    ],
    transform: {
        '^.+.(ts|mjs|js|html)$': [
            'jest-preset-angular',
            {
                stringifyContentPathRegex: '\\.(html|svg)$',

                tsconfig: '<rootDir>/tsconfig.spec.json',
            },
        ],
    },
    transformIgnorePatterns: ['node_modules/(?!.*.mjs$)'],
};
