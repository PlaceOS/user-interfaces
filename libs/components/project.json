{
    "name": "components",
    "$schema": "../../node_modules/nx/schemas/project-schema.json",
    "projectType": "library",
    "sourceRoot": "libs/components/src",
    "prefix": "placeos",
    "targets": {
        "lint": {
            "executor": "@nx/eslint:lint",
            "outputs": ["{options.outputFile}"]
        },
        "test": {
            "executor": "@nx/jest:jest",
            "outputs": ["{workspaceRoot}/coverage/libs/components"],
            "options": {
                "jestConfig": "libs/components/jest.config.ts"
            }
        }
    },
    "tags": []
}
