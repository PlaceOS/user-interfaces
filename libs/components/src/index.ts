export * from './lib/components.module';

export { AuthorisedAdminGuard } from './lib/guards/authorised-admin.guard';
export { AuthorisedUserGuard } from './lib/guards/authorised-user.guard';
