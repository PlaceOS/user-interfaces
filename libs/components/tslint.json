{
    "extends": "../../tslint.json",
    "rules": {
        "directive-selector": [
            true,
            "attribute",
            "userInterfaces",
            "camelCase"
        ],
        "component-selector": [true, "element", "user-interfaces", "kebab-case"]
    },
    "linterOptions": {
        "exclude": ["!**/*"]
    }
}
