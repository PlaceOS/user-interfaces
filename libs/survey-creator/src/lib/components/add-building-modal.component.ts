import { Component, OnInit } from '@angular/core';
import { MatDialog, MatDialogRef } from '@angular/material/dialog';

@Component({
    selector: 'add-building-modal',
    template: `
        <section>
            <div class="header">
                <span class="dialog-title">Add Building</span>
                <span class="close" (click)="closeDialog()">
                    <mat-icon
                        aria-hidden="false"
                        aria-label="Material icon for closing dialog"
                        class="icon"
                        >close</mat-icon
                    >
                </span>
            </div>
            <main>
                <span class="small-title">Building image</span>
                <div class="upload-container"></div>
                <span class="small-title">Building name</span>

                <span class="small-title">Building location</span>
            </main>
            <footer>
                <button
                    mat-button
                    class="cancel-button"
                    color="basic"
                    (click)="closeDialog()"
                >
                    Cancel
                </button>
                <button
                    mat-button
                    class="add-button"
                    color="primary"
                    (click)="addBuilding()"
                >
                    Add
                </button>
            </footer>
        </section>
    `,
    styles: [
        `
            .header {
                display: flex;
                align-items: center;
                justify-content: space-between;
                background-color: #fff;
                height: 70px;
                border: 1px solid rgba(0, 0, 0, 0.12);
            }
            main {
                margin: 20px;
                display: flex;
                flex-direction: column;
            }
            .dialog-title {
                display: flex;
                font-size: 20px;
                line-height: 30px;
                font-weight: 800;
                padding: 20px;
            }
            .small-title {
                font-size: 14px;
                line-height: 24px;
                font-weight: 700;
                margin: 8px 0px 8px 0px;
            }
            .upload-container {
                background-color: #f8f8fa;
                width: 380px;
                height: 259px;
                border: 1px solid #d4d4d4;
                border-radius: 6px;
            }
            .cancel-button {
                display: flex;
                color: #292f5b;
                background-color: #fff;
                border-radius: 2px;
                margin: 20px 0px 20px 20px;
            }
            .add-button {
                display: flex;
                color: #fff;
                background-color: #292f5b;
                border-radius: 2px;
                margin: 20px 0px 20px 20px;
            }
            footer {
                display: flex;
                flex-direction: row;
                justify-content: flex-end;
                margin-right: 20px;
            }
            .close {
                display: flex;
                z-index: 99;
                text-align: right;
                vertical-align: middle;
                line-height: 12px;
                margin-right: 20px;
            }
        `,
    ],
})
export class AddBuildingModalComponent implements OnInit {
    constructor(public dialogRef: MatDialogRef<AddBuildingModalComponent>) {}

    ngOnInit(): void {}

    closeDialog() {
        this.dialogRef.close();
    }

    addAnotherBuilding(): void {
        return;
    }

    addBuilding(): void {
        return;
    }
}
