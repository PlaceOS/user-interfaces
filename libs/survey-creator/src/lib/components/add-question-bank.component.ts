import { Component, OnInit, OnDestroy } from '@angular/core';
import { MatDialog, MatDialogRef } from '@angular/material/dialog';
import { FormControl } from '@angular/forms';
import { BehaviorSubject, Subscription } from 'rxjs';
import { Question, QuestionType } from '../survey-types';
import { SurveyCreatorService } from '../survey-creator.service';

@Component({
    selector: 'add-question-bank',
    template: `
        <section>
            <div class="header">
                <span class="dialog-title">Add to question bank</span>
                <span class="close" (click)="closeDialog()">
                    <mat-icon
                        aria-hidden="false"
                        aria-label="Material icon for closing dialog"
                        class="icon"
                        >close</mat-icon
                    >
                </span>
            </div>
            <main>
                <div><span class="section-heading">Category</span></div>
                <div>
                    <p class="small-text">
                        Please select tags that apply to the below question.
                    </p>
                </div>
                <div class="category-tags" *ngFor="let tag of tags">
                    <mat-checkbox color="primary">{{ tag }}</mat-checkbox>
                </div>
                <div class="question-box">
                    <div class="new-question">
                        <ng-container
                            [ngTemplateOutlet]="
                                question_type == QuestionType.text
                                    ? Text
                                    : question_type == QuestionType.checkbox
                                    ? Checkbox
                                    : question_type == QuestionType.comment
                                    ? Comment
                                    : question_type == QuestionType.dropdown
                                    ? Dropdown
                                    : Rating
                            "
                        >
                        </ng-container>
                        <ng-template #Rating>
                            <rating-question
                                [question]="placeholder_question_text"
                                [view]="'draft'"
                                [preview]="true"
                                (newTitleEvent)="
                                    updateTitle($event, QuestionType.rating)
                                "
                                (newRatingEvent)="updateRating($event)"
                            ></rating-question>
                        </ng-template>
                        <ng-template #Text>
                            <text-question
                                [question]="placeholder_question_text"
                                [view]="'draft'"
                                [preview]="true"
                                (newTitleEvent)="
                                    updateTitle($event, QuestionType.text)
                                "
                            ></text-question>
                        </ng-template>
                        <ng-template #Comment>
                            <comment-box-question
                                [question]="placeholder_question_text"
                                [view]="'draft'"
                                [preview]="true"
                                (newTitleEvent)="
                                    updateTitle($event, QuestionType.comment)
                                "
                            ></comment-box-question>
                        </ng-template>
                        <ng-template #Checkbox>
                            <checkbox-question
                                [question]="placeholder_question_select"
                                [view]="'draft'"
                                [preview]="true"
                                (newTitleEvent)="
                                    updateTitle($event, QuestionType.checkbox)
                                "
                                (allChoicesEvent)="updateAllChoices($event)"
                            ></checkbox-question>
                        </ng-template>
                        <ng-template #Dropdown>
                            <dropdown-question
                                [question]="placeholder_question_select"
                                [view]="'draft'"
                                [preview]="true"
                            ></dropdown-question>
                        </ng-template>
                    </div>
                    <section class="options-container">
                        <div class="dropdown-container">
                            <mat-form-field appearance="none" class="dropdown">
                                <mat-select
                                    [(value)]="selected_tag"
                                    (selectionChange)="updateSelectedTag()"
                                >
                                    <mat-option
                                        *ngFor="
                                            let enum of QuestionType | keyvalue
                                        "
                                        [value]="enum.value"
                                    >
                                        {{ enum.value }}
                                    </mat-option>
                                </mat-select>
                            </mat-form-field>
                        </div>

                        <div class="required-container">
                            <mat-slide-toggle> </mat-slide-toggle>
                            <span>Required</span>
                        </div>
                    </section>
                </div>
                <div>
                    <button
                        mat-button
                        class="add-another-question"
                        (click)="addAnotherQuestion()"
                    >
                        Add another
                    </button>
                </div>
            </main>
            <footer>
                <button
                    mat-button
                    class="cancel-button"
                    color="basic"
                    (click)="closeDialog()"
                >
                    Cancel
                </button>
                <button
                    mat-button
                    class="add-button"
                    color="primary"
                    (click)="addQuestion()"
                >
                    Add
                </button>
            </footer>
        </section>
    `,
    styles: [
        `
            ::ng-deep .mat-select-value {
                max-width: min-content;
                margin-right: 3px;
            }
            .header {
                display: flex;
                align-items: center;
                justify-content: space-between;
                background-color: #fff;
                height: 70px;
                border: 1px solid rgba(0, 0, 0, 0.12);
            }
            main {
                margin: 20px;
            }
            .dialog-title {
                display: flex;
                font-size: 20px;
                line-height: 30px;
                font-weight: 800;
                padding: 20px;
            }
            .section-heading {
                font-size: 16px;
                font-weight: 700;
                line-height: 24px;
            }
            .small-text {
                size: 11px;
                font-weight: 400;
                margin-top: 3px;
                color: #808080;
            }
            .category-tags {
                display: inline-flex;
            }

            .category-tags mat-checkbox {
                margin: 10px 40px 10px 0px;
            }
            .question-box {
                border: 1px solid #3b82f6;
                box-shadow: 0px 0px 1px rgba(15, 23, 42, 0.06),
                    0px 20px 25px -5px rgba(15, 23, 42, 0.1),
                    0px 10px 10px -5px rgba(15, 23, 42, 0.04);
                border-radius: 4px;
                margin-top: 20px;
                padding: 10px;
            }
            .new-question {
                margin-left: -10px;
            }
            .dropdown-container {
                display: inline-flex;
                position: relative;
                width: 100px;
                margin-bottom: -10px;
            }
            .dropdown {
                display: absolute;
                font-size: 12px;
                width: 100%;
                height: 30px;
                margin-bottom: 10px;
            }
            .options-container {
                display: inline-flex;
                width: 100%;
                justify-content: flex-end;
                align-items: center;
            }
            .required-container {
                display: inline-flex;
                font-size: 12px;
                justify-content: space-between;
                width: 90px;
                align-items: center;
                margin: 0px 15px 5px 0px;
            }
            .add-another-question {
                background-color: #eeeeeeee;
                width: 100%;
                border-radius: 0px;
                color: black;
                border: none;
                margin-top: 30px;
            }
            .cancel-button {
                display: flex;
                color: #292f5b;
                background-color: #fff;
                border-radius: 2px;
                margin: 20px 0px 20px 20px;
            }
            .add-button {
                display: flex;
                color: #fff;
                background-color: #292f5b;
                border-radius: 2px;
                margin: 20px 0px 20px 20px;
            }
            footer {
                display: flex;
                flex-direction: row;
                justify-content: flex-end;
                margin-right: 20px;
            }
            .close {
                display: flex;
                z-index: 99;
                text-align: right;
                vertical-align: middle;
                line-height: 12px;
                margin-right: 20px;
            }
        `,
    ],
})
export class AddQuestionBankComponent implements OnInit {
    tags: string[] = ['Desk', 'Room', 'Parking'];
    selected_tag: any = QuestionType.rating;
    question_type: string = QuestionType.rating;
    new_question: Question;
    new_questions: Question[] = [];
    private _update_flag: BehaviorSubject<boolean> =
        new BehaviorSubject<boolean>(false);
    flag_sub: Subscription;
    placeholder_choice: string = 'Type a choice here...';
    placeholder_question_text: Question = {
        type: QuestionType.text || QuestionType.comment || QuestionType.rating,
        title: 'Type a question...',
        name: '',
    };
    placeholder_question_select: Question = {
        type: QuestionType.checkbox || QuestionType.dropdown,
        title: 'Type a question...',
        name: '',
        choices: [this.placeholder_choice],
    };

    public QuestionType = QuestionType;

    constructor(
        public dialogRef: MatDialogRef<AddQuestionBankComponent>,
        private _surveyCreatorService: SurveyCreatorService
    ) {}

    ngOnInit(): void {
        console.log(this.question_type, 'question type');
        console.log(this.selected_tag, 'selected_tag');

        this.new_question = {
            title: '',
            type: '' as any,
            name: '',
        };
    }

    updateSelectedTag() {
        this.question_type = this.selected_tag;
    }

    closeDialog() {
        this.dialogRef.close();
    }

    updateTitle(event, question_type) {
        this.new_question.title = event;
        this.new_question.type = question_type;
    }

    updateChoice(event) {
        let choice_index = this.new_question.choices.findIndex(
            (item) => item == event[1]
        );
        if (choice_index) {
            this.new_question.choices[choice_index] = event[0];
        }
    }

    updateAllChoices(event: string[]) {
        this.new_question.choices = event;
    }
    updateRating(event) {
        this.new_question.rateValues = event || [1, 2, 3, 4, 5];
    }

    addAnotherQuestion() {
        console.log('add');
    }

    addQuestion() {
        // switch (this.new_question.type) {
        //     case QuestionType.rating:
        //         this.new_question.title = event[0];
        //         this.new_question.rateValues = event[1];

        //     case QuestionType.checkbox || QuestionType.dropdown:
        //         console.log(event, 'event in Add modal');
        //         this.new_question.title = event[0];
        //         this.new_question.choices = event[1];
        // }
        this._update_flag.next(true);
        this.closeDialog();
        console.log(this.new_question, 'q to be added');
        this.flag_sub = this._update_flag.asObservable().subscribe((flag) => {
            if (flag) {
                this._surveyCreatorService.question_bank.push(
                    this.new_question
                );
            }
        });
        this._update_flag.next(false);
    }

    ngOnDestroy(): void {
        this.flag_sub?.unsubscribe();
    }
}
