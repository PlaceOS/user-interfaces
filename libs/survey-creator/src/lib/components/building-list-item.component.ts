import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Building } from '@placeos/organisation';
import { BuildingsService } from '../buildings.service';
import { Router, ActivatedRoute } from '@angular/router';

@Component({
    selector: 'building-list-item',
    template: `
        <section class="building-wrapper">
            <div
                class="options"
                [matMenuTriggerFor]="optionsMenu"
                aria-label="more options"
            >
                <mat-icon>more_horiz</mat-icon>
            </div>
            <mat-menu #optionsMenu="matMenu">
                <div class="menu-wrapper">
                    <button mat-menu-item (click)="editBuilding()">
                        <mat-icon>edit</mat-icon>
                        <span>Edit</span>
                    </button>
                    <button mat-menu-item (click)="deleteBuilding()">
                        <mat-icon>delete_forever</mat-icon>
                        <span>Delete</span>
                    </button>
                </div>
            </mat-menu>
            <div *ngIf="building.image" class="image-container">
                <img
                    [src]="building.image"
                    alt="image of building"
                    height="230"
                    width="350"
                />
            </div>
            <div *ngIf="!building.image" class="image-container">
                <span> <mat-icon class="domain-icon">domain</mat-icon></span>
            </div>
            <div class="details-container">
                <div class="location-wrapper">
                    <mat-icon
                        class="location-icon"
                        aria-hidden="false"
                        aria-label="Material icon of location pointer"
                        >location_on</mat-icon
                    >
                    <span> {{ building.address }} </span>
                </div>

                <span class="building-title">
                    {{ building.name }}
                </span>
                <ul class="details-text">
                    <li>surveys live</li>
                    <li>draft</li>
                    <li>responses</li>
                </ul>
            </div>
            <div class="button-container">
                <button
                    mat-button
                    class="view-button"
                    color="basic"
                    (click)="navigate()"
                >
                    <span>View</span>

                    <span>
                        <mat-icon class="right_arrow"
                            >chevron_right</mat-icon
                        ></span
                    >
                </button>
            </div>
        </section>
    `,
    styles: [
        `
            .building-wrapper {
                display: flex;
                position: relative;
                flex-direction: row;
                position: relative;
                justify-content: flex-start;
                height: 260px;
                min-width: 1100px;
                width: 80%;
                margin: 25px 200px;
                border: 1px solid #e6e6e6;
                box-shadow: 0px 2px 4px rgba(5, 28, 44, 0.1);
                border-radius: 6px;
            }
            .options {
                display: flex;
                align-items: center;
                justify-content: center;
                background-color: #fff;
                position: absolute;
                height: 48px;
                width: 48px;
                top: 10px;
                left: 10px;
                border: 1px solid #e6e6e6;
                border-radius: 3px;
            }
            .menu-wrapper {
                display: flex;
                flex-direction: column;
                height: 90px;
                width: 135px;
                justify-content: center;
            }
            .image-container {
                display: flex;
                height: 100%;
                width: 380px;
                background-color: rgba(0, 0, 0, 0.12);
                align-items: center;
                justify-content: center;
            }
            .details-container {
                display: flex;
                flex-direction: column;
                min-width: 600px;
                margin: 30px;
            }
            .location-wrapper {
                display: flex;
                flex-direction: row;
                align-items: center;
                margin-bottom: 20px;
            }
            .location-icon {
                color: #5295f7;
                margin-right: 10px;
            }
            .domain-icon {
                display: flex;
                color: #fff;
                height: 100%;
                width: 100%;
            }
            .building-title {
                font-size: 18px;
                font-weight: 500;
            }
            .details-text {
                display: flex;
                flex-direction: column;
                font-size: 14px;
                font-weight: 400;
                color: #020202;
                margin-bottom: 10px;
                justify-content: space-between;
                height: 70px;
            }
            .button-container {
                display: flex;
                position: absolute;
                bottom: 20px;
                right: 20px;
            }
            .view-button {
                display: inline-flex;
                color: #292f5b;
                background-color: #fff;
                border-radius: 2px;
                margin: 20px 0px 20px 20px;
                justify-content: center;
                padding-left: 20px;
            }
            .view-button span {
                display: inline-flex;
            }
            .view-button mat-icon {
                display: flex;
                justify-content: center;
                align-items: center;
            }
        `,
    ],
})
export class BuildingListItemComponent implements OnInit {
    @Input() building: Building | any;
    // @Output() deleteBuildingEvent = new EventEmitter<any>();

    constructor(
        public buildingsService: BuildingsService,
        public router: Router,
        public route: ActivatedRoute
    ) {}

    ngOnInit(): void {}

    editBuilding() {}

    deleteBuilding() {
        this.buildingsService.deleteBuilding(this.building);
    }

    viewBuilding() {}

    navigate(): void {
        this.router.navigate(['survey-list', this.building.name], {
            relativeTo: this.route,
        });
    }
}
