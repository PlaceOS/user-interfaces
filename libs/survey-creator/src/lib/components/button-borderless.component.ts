import { Component, OnInit, Input } from '@angular/core';

@Component({
    selector: 'button-borderless',
    template: `
        <button class="button-with-icon">
            <span class="">
                <mat-icon
                    aria-hidden="false"
                    aria-label="Material icon for button"
                    class="icon"
                    [ngClass]="icon_color == 'red' ? 'red' : 'blue'"
                    >{{ icon }}</mat-icon
                >
            </span>
            <span
                class="button-title"
                [ngClass]="text_color == 'black' ? 'black' : 'blue'"
            >
                {{ button_title }}
            </span>
        </button>
    `,
    styles: [
        `
            .button-with-icon {
                height: 35px;
                border: none;
                font-size: 12px;
                background-color: #fff;
                margin: 10px 0px 10px 10px;
                display: inline-flex;
                align-items: center;
                flex-direction: row;
                text-decoration: none;
                padding: 0px 10px 0px 0px;
            }

            .button-with-icon:hover {
                background-color: #f5f5f5;
            }

            .button-with-icon:active {
                background-color: #f5f5f5;
            }
            .button-with-icon span {
                display: inline-flex;
                align-items: center;
                vertical-align: middle;
            }
            .icon {
                display: flex;
                align-items: center;
                justify-content: center;
                margin-right: 5px;
            }
            .blue {
                color: #3e69cf;
            }
            .red {
                color: #9a2d2d;
            }
            .grey {
                color: #808080;
            }
            .black {
                color: #000000;
            }
            .button-title {
                display: inline-flex;
                vertical-align: middle;
            }
        `,
    ],
})
export class ButtonBorderlessComponent implements OnInit {
    @Input() button_title: string = '';
    @Input() icon?: string = '';
    @Input() icon_color: string = '';
    @Input() text_color: string = '';
    constructor() {}

    ngOnInit(): void {
        console.log(this.icon_color, 'icon color');
    }
}
