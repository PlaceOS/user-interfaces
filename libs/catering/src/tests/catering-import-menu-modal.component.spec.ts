import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { IconComponent } from '@placeos/components';
import { MockComponent } from 'ng-mocks';

import { CateringImportMenuModalComponent } from '../lib/catering-import-menu-modal.component';

describe('CateringImportMenuModalComponent', () => {
    let spectator: Spectator<CateringImportMenuModalComponent>;
    const createComponent = createComponentFactory({
        component: CateringImportMenuModalComponent,
        declarations: [MockComponent(IconComponent)],
        imports: [MatProgressSpinnerModule],
    });

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(spectator.element).toMatchSnapshot();
    });

    it('should allow downloading the template', () => {
        const spy = jest.spyOn(spectator.component, 'downloadTemplate');
        spectator.click('main button');
        expect(spectator.component.downloadTemplate).toHaveBeenCalled();
    });
});
