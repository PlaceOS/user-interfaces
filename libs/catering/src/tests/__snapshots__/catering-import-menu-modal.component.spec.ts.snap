// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CateringImportMenuModalComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <header
    class="h-12 bg-primary flex items-center justify-between px-4"
  >
    <h2>
      Import Catering Menu
    </h2>
    <button
      mat-dialog-close=""
      mat-icon-button=""
    >
      <app-icon>
        close
      </app-icon>
    </button>
    
  </header>
  <main>
    <div
      class="relative flex flex-col items-center justify-center space-y-2 h-[24rem] w-[24rem] border-4 border-gray-200 border-dashed rounded-xl hover:bg-gray-100 m-4 p-4 cursor-pointer"
    >
      <app-icon
        class="text-4xl"
      >
        upload
      </app-icon>
      <p>
        Click to select file or Drag and drop files
      </p>
      <input
        class="absolute inset-0 opacity-0"
        type="file"
      />
    </div>
    <div
      class="flex items-center justify-center px-4 pb-4"
    >
      <button
        class="clear underline w-48"
        mat-button=""
      >
         Download Template 
      </button>
    </div>
  </main>
  
  
</div>
`;
